### ISSUE TITLE
Outdated assumption about releases folder breaks working with newer boshreleases

### ISSUE THREAD

**Issue by voelzmo** [cloudfoundry-incubator/bosh-workspace#26]
Since version 2682 the BOSH CLI creates a subfolder with with release's name in the `releases` folder: https://groups.google.com/a/cloudfoundry.org/forum/#!search/final$20release$20directory/bosh-users/W3A3OIeYN_E/yL0dIS0_V7gJ
This results in a structure where the actual `foo-<version>.yml` files are sitting in `releases/foo/foo-<version>.yml` instead of just `releases/foo-<version>.yml`

Currently, bosh-workspace only supports the old structure and cannot find the desired final version for boshreleases built with the new structure.

Some boshreleases, have worked around this by making `releases/foo` a symlink pointing to `releases`, see e.g. [cf-release](https://github.com/cloudfoundry/cf-release/blob/master/releases/cf). However, if you don't control the boshrelease yourself, you cannot use bosh-workspace with it right now.

**Comment by rkoster** [comment]
@voelzmo thanks for raising this issue. I my self indeed worked around this by creating a symlink. But this issue is indeed something which should be fixed in the bosh-workspace.

**Comment by voelzmo** [comment]
@rkoster If I knew how to update the spec assets you're using in there, I could take a shot at this myself. However, you seem to include a zipped version of a git repo - I'm not sure how I could include an example repository with a different structure for the `releases` folder. Could you provide some hints?

**Comment by rkoster** [comment]
Yhea I should create a nice rake task for building those assets but what I usually do is:
- Unzip the asset dir
- Make some changes with git
- zip the `.git` directory (`zip -r ../../foo-boshrelease-repo.zip .git`)

**Comment by rkoster** [comment]
One more thing, if you are gonna implement a fix could you do this on top of the [deployment-pach](https://github.com/cloudfoundry-incubator/bosh-workspace/tree/deployment-patch) branch since this will soon be the new master (I'm waiting for our dev openstack env to be up again to do a full integration test). Since in this branch the ruby git gem was removed with [rugged](https://github.com/libgit2/rugged).

**Comment by voelzmo** [comment]
@rkoster When implementing this, I really struggled with a few non-obvious thing. Creating the new test files wasn't easy, but thanks to your help I had a good starting point. Still, for someone else who hasn't talked to you, this is a huge obstacle. Second, I found the repository update logic to be a real problem, that's why I made an additional change to always checkout HEAD if no other ref has been given.

Currently, you look for the newest file in the releases folder and checkout the revision with the last commit to that file - this doesn't cope well with changes in the directory structure only. This was a bit implicit in the code and not very obvious for me. Let me know what you think about the change!

Warm regards
Marco
